connect/resource/usage_automation.py

```diff
@@ -0,0 +1,172 @@
+            # Could be because description is empty or None, so make sure it is empty
+            usage_file.description = ''
+        response = self.api.post(self.url, data=usage_file)
+        return self._load_schema(response, many=False)
+
+    @staticmethod
+    def create_spreadsheet(usage_records):
+        # type: (List[FileUsageRecord]) -> openpyxl.Workbook
+        book = openpyxl.Workbook()
+        sheet = book.active
+        sheet.title = 'usage_records'
+        sheet['A1'] = 'record_id'
+        sheet['B1'] = 'item_search_criteria'
+        sheet['C1'] = 'item_search_value'
+        sheet['D1'] = 'quantity'
+        sheet['E1'] = 'start_time_utc'
+        sheet['F1'] = 'end_time_utc'
+        sheet['G1'] = 'asset_search_criteria'
+        sheet['H1'] = 'asset_search_value'
+        for index, record in enumerate(usage_records):
+            row = str(index + 2)
+            sheet['A' + row] = record.record_id
+            sheet['B' + row] = record.item_search_criteria
+            sheet['C' + row] = record.item_search_value
+            sheet['D' + row] = record.quantity
+            sheet['E' + row] = record.start_time_utc
+            sheet['F' + row] = record.end_time_utc
+            sheet['G' + row] = record.asset_search_criteria
+            sheet['H' + row] = record.asset_search_value
+        return book
+
+    def upload_spreadsheet(self, usage_file, spreadsheet):
+        # type: (File, openpyxl.Workbook) -> None
+
+        # Generate spreadsheet file
+        with NamedTemporaryFile() as tmp:
+            spreadsheet.save(tmp)
+            tmp.seek(0)
+            file_contents = tmp.read()
+
+        # Setup request
+        url = self.urljoin(self.url, usage_file.id, 'upload/')
+        headers = self.api.headers
+        headers['Accept'] = 'application/json'
+        del headers['Content-Type']  # This must NOT be set for multipart post requests
+        multipart = {'usage_file': ('usage_file.xlsx', file_contents)}
+        logger.info('HTTP Request: {} - {} - {}'.format(url, headers, multipart))
+
+        # Post request
+        try:
+            response = requests.post(url, headers=headers, files=multipart)
+        except requests.RequestException as ex:
+            raise FileCreationError('Error uploading file: {}'.format(ex))
+        logger.info('HTTP Code: {}'.format(response.status_code))
+        if response.status_code != 201:
+            msg = 'Unexpected server response, returned code {}'.format(response.status_code)
+            logger.error('{} -- Raw response: {}'.format(msg, response.content))
+            raise FileCreationError(msg)
+
+    def upload_usage_records(self, usage_file, usage_records):
+        # type: (File, List[FileUsageRecord]) -> None
+        # TODO: Using xslx mechanism till usage records json api is available
+        book = self.create_spreadsheet(usage_records)
+        self.upload_spreadsheet(usage_file, book)
+
+    def get_usage_template(self, product):
```

Comment:
Returning string looks strange ... binary blob in a string ...
probably operate with file object?
How to use it - need example, like:

with open("template.xlsx", "w") as ssf:
   ssf.write(UsageAutomation().get_usage_template('PRD-xxx-yyy-zzz'))

Reply:
I have been checking this and you are right. A "bytes" object should be returned.

It works like this:

- The API '/usage/products/{product_id}/template/' is used to retrieve the link to the usage template (I just spotted a bug in the code here, the 'template' part is missing; will be fixed in next commit).
- We try to read the contents of the location provided. The contents of the file are returned as text, but since it will be an xslx file, it makes more sense to use the "content" field of the response rather than the "text" field, and obtain it as a bytes object.

Apart from using requests to obtain the file contents, a simple file read is also attempted, in case that the file is stored locally. I think that this should actually only happen during testing, and since there are other alternatives to achieve this during testing (by mocking the requests.get function to return the contents of a file), I didn't come up with the best solution here. I will remove the part where the file is loaded and use requests only.

Reply:
Done,